network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -407,34 +407,65 @@ public void deleteNetwork(UUID uuid, int variantNum) {
         }
     }
 
-    @SuppressWarnings("javasecurity:S5145")
-    public void cloneNetwork(UUID uuid, int sourceVariantNum, int targetVariantNum, String targetVariantId) {
+    public void cloneNetwork(UUID uuid, UUID sourceNetworkUuid, List<String> targetVariantIds) {
+        List<VariantInfos> networkVariantsInfo = getVariantsInfos(sourceNetworkUuid);
+        networkVariantsInfo.stream().filter(v -> targetVariantIds.contains(v.getId())).forEach(variantInfos -> {
+            Resource<NetworkAttributes> sourceNetworkAttribute = getNetwork(sourceNetworkUuid, variantInfos.getNum()).orElse(null);
```

Comment:
Use elseThrow(...) so that you don't need after to check for null et throw an exception.